### PR TITLE
Bug 2029438: Set rc-manager=unmanaged on baremetal bootstrap

### PR DESCRIPTION
**Pull request by cybertron**
There is a dispatcher script on the bootstrap node that is supposed
to be responsible for updating resolv.conf on network changes. We
don't want NetworkManager to step on those changes, so it needs to
be set to unmanaged. This is already the case on cluster nodes[0],
but apparently was missed in the bootstrap config.

0: https://github.com/openshift/machine-config-operator/blob/4b98289781ec5f3fe7313049496afda654f6e0fc/templates/common/on-prem/files/NetworkManager-onprem.conf.yaml#L7

### CODE REVIEW

**Review by staebler — COMMENTED** [review]
Is this change specific to baremetal since that is the only platform that includes a conf for NetworkManager? Or do the other on-prem platforms (openstack, ovirt, vsphere) that include a local-dns-prepender dispatcher need to include a conf with this sections as well?

**Comment by yboaron** [comment]
Q: Is updating NM rc-manager to unmanaged is enough?  shouldn't we also update the prepender [1]  script (similar to node's prepended [2] ) ? 

[1] https://github.com/openshift/installer/blob/b8e2c497249948adffba128dac0713b521b3ce4a/data/data/bootstrap/baremetal/files/etc/NetworkManager/dispatcher.d/30-local-dns-prepender#L21-#L29

[2] 
https://github.com/openshift/machine-config-operator/blob/master/templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml#L70

**Comment by cybertron** [comment]
/hold

Okay, yeah this needs more changes. I'll make sure to apply the eventual fix to the other platforms as well.

**Comment by cybertron** [comment]
/hold cancel

In my local test this did the right thing so assuming it passes ci on the other platforms we should be good.

**Review by staebler — COMMENTED** [review]
Looks sane to me, although the intricacies are out of my realm of expertise.
/approve

I am interested in ways that we can reduce the duplication across the platforms and between the installer and the MCO.

File: data/data/bootstrap/vsphere/files/etc/NetworkManager/conf.d/99-ovirt.conf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[main]
+rc-manager=unmanaged
```
Comment (mandre) [inline data/data/bootstrap/vsphere/files/etc/NetworkManager/conf.d/99-ovirt.conf:2]: This file should be named `data/data/bootstrap/vsphere/files/etc/NetworkManager/conf.d/99-vsphere.conf`

**Review by mandre — COMMENTED** [review]
Just a small nit about file name, otherwise LGTM from an openstack point of view.

**Comment by patrickdillon** [comment]
@cybertron there's one small change above: https://github.com/openshift/installer/pull/5482#discussion_r817448183

Can you fix that so we can move this through? I'll try to get some eyes from baremetal

@rvanderp3 @jcpowermac  PTAL for vsphere sanity checking

**Comment by jcpowermac** [comment]
/test e2e-vsphere 

(last run was in december)

**Comment by jcpowermac** [comment]
> @cybertron there's one small change above: [#5482 (comment)](https://github.com/openshift/installer/pull/5482#discussion_r817448183)
> 
> Can you fix that so we can move this through? I'll try to get some eyes from baremetal
> 
> @rvanderp3 @jcpowermac PTAL for vsphere sanity checking

Seem fine to me since we already are doing this in the other cluster nodes.

**Comment by patrickdillon** [comment]
/lgtm cancel

ready to merge except the misnamed file

File: data/data/bootstrap/vsphere/files/etc/NetworkManager/conf.d/99-ovirt.conf (hunk 1 above)
Comment (cybertron) [reply data/data/bootstrap/vsphere/files/etc/NetworkManager/conf.d/99-ovirt.conf:2]: Oops! Fixed.

**Comment by cybertron** [comment]
> I am interested in ways that we can reduce the duplication across the platforms and between the installer and the MCO.

In MCO we created a separate "on-prem" directory that applies the same config to all on-prem platforms. I don't _think_ we're able to eliminate the duplication between MCO and installer, but that would reduce the duplication in the installer itself.

**Review by mandre — COMMENTED** [review]
/lgtm

**Comment by mandre** [comment]
Seems like flakes to me.
/retest

**Comment by cybertron** [comment]
/retest-required

metal hit the etcd-quorum-guard bug, aws seems to have quota issues.

https://bugzilla.redhat.com/show_bug.cgi?id=2063831

**Comment by patrickdillon** [comment]
/hold cancel
Jobs are looking healthy

**Comment by cybertron** [comment]
/skip

This seems to be blocking on e2e-metal-ipi-ovn-ipv6-required, which appears to have been replaced by just ovn-ipv6 that has passed.

**Comment by cybertron** [comment]
Okay, that didn't work either. The job isn't even being re-triggered so I guess we need someone to override it?

**Comment by patrickdillon** [comment]
/override ci/prow/e2e-metal-ipi-ovn-ipv6

This test has passed twice but is failing due to infra issues.

**Pull request merged**